torch/_subclasses/meta_utils.py

```diff
@@ -442,6 +445,24 @@ def empty_create(inner_t, inner_src):
                                 )
                             ),
                         )
+                        if r.is_nested:
+                            assert not is_symbolic(r._size[1])
+                            # Avoid circular import
+                            from torch._dynamo.source import (
+                                TensorProperty,
+                                TensorPropertySource,
+                            )
+
+                            sym_ragged_size = shape_env.create_symintnode(
+                                shape_env.create_symbol(
+                                    r._size[1],
+                                    TensorPropertySource(
+                                        source, TensorProperty.SIZE, 1
+                                    ),
+                                ),
+                                hint=r._size[1],
+                            )
+                            r.set_raggedness_id(sym_ragged_size)
```

Comment:
What's this?

Reply:
With the current fakification logic
- (1) we fakify the inner tensors and pass them to `__torch_unflatten__`
- (2) user is responsible for reconstruct the outer subclass, e.g. construct the sizes of outer tensor using whatever symints are available from the inner tensors

Sometimes it is possible to derive the outer sizes from the inner ones: for example, I can compute the symint for the `B` in `[B, *, D]` if I do `offsets.shape[0] - 1`.

In the case of the ragged dimension, this is not possible, however, and so we need to do something special.

There are probably a few options here, adding this `set_raggedness_id` method to smash the symbolic raggedness dimension onto the wrapper subclass is one of them. Alternatively, we could've extended __tensor_unflatten__ to accept an additional `source` argument, and constructed the raggedness symint from inside there.

I don't have a strong opinion on which option which pick here, but the idea was to avoid modifying the common `__tensor_unflatten__` API given that we don't have any other examples of subclasses to go off of.